options.go

```diff
@@ -127,6 +128,9 @@ type Options struct {
 	// Don't enable it unless you are 100% sure what you are doing.
 	// Note that this option is only available for stream-oriented protocol.
 	EdgeTriggeredIO bool
+
+	// TLSConfig support TLS
+	TLSConfig *tls.Config
```

Comment:
Make this an interface instead of a pointer of a specific struct, that way we can implement TLS outside the `gnet` and pass the interface to the `gnet`. Decoupling the TLS from `gnet` also enables the users to use other TLS implementations that implement the interface of `gnet` TLS.

Reply:
> Make this an interface instead of a pointer of a specific struct, that way we can implement TLS outside the `gnet` and pass the interface to the `gnet`. Decoupling the TLS from `gnet` also enables the users to use other TLS implementations that implement the interface of `gnet` TLS.

 Looking forward to this merge. This pr is great !  @leslie-fei ，Could you please modify this according to  @panjf2000  's requirements?